datavault-common/src/main/java/org/datavaultplatform/common/util/RoleUtils.java

```diff
@@ -12,4 +19,15 @@ private RoleUtils() {}
     public static boolean isReservedRoleName(String roleName) {
         return IS_ADMIN_ROLE_NAME.equalsIgnoreCase(roleName) || DATA_OWNER_ROLE_NAME.equalsIgnoreCase(roleName);
     }
+
+    public static boolean hasReducedPermissions(Collection<PermissionModel> originalPermissions,
+                                                Collection<PermissionModel> newPermissions) {
+        Set<Permission> original = originalPermissions.stream()
```

Comment:
We're only comparing the permissions of the roles that were updated, is that right?

Reply:
It was right, but not any more. This change originally only cared if permissions were downgraded so we only needed to check the updates here, but we got confirmation yesterday that we also want to force logout any users that gain new permissions so this bit of code will be gotten rid of shortly

Reply:
Ah, nice. Makes it simpler.